### PR TITLE
Add zone id to either worker route endpoint

### PR DESCRIPTION
**Pull request by jafowler**
i could be wrong, but after running some tests on this i get the feeling that `cloudflare_workers_routes` is broken for some using the `routes` endpoint. It will break for anyone who supplies an AccountID. 

Before:
```
resource "cloudflare_worker_route" "terraform_managed_resource" {
  pattern     = "example.com/*"
  script_name = "my-script"
}
```
After: 
```
resource "cloudflare_worker_route" "terraform_managed_resource" {
  pattern     = "example.com/*"
  script_name = "my-script"
  zone_id     = "0da42c8d2132a9ddaf714f9e7c920711"
}
```

### CODE REVIEW

File: internal/app/cf-terraforming/cmd/generate_test.go [hunk 1]
```diff
@@ -148,7 +149,17 @@ func TestResourceGeneration(t *testing.T) {
 			storedAccountID := viper.GetString("account")
 			storedZoneID := viper.GetString("zone")
 
-			if tc.identiferType == "account" {
+			// edge case for the weird workflow that is cloudflare_worker_routes
+			if tc.testdataFilename == "cloudflare_worker_route_routes" {
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate_test.go:153]: this feels like something is really wrong here. we shouldn't be basing this logic on specific test cases (especially here when we have conflicting flags `--zone` and `--account`). we should only have one or the other.

File: internal/app/cf-terraforming/cmd/generate.go [hunk 2]
```diff
@@ -634,9 +634,21 @@ func generateResources() func(cmd *cobra.Command, args []string) {
...
+				remap "script_name" to the "script" value.
+				Add zone_id to each resource as well.
+
+				The reason we're adding zone_id here is that its a required field.
+				Its needs to be included when using either `filter` or `routes` endpoint.
+				Currently, we don't add it when using the `routes` endpoint due to logic below
+				where we only add `zone_id` if an `accountID` is not available (added for page_rules)
+				https://github.com/cloudflare/cf-terraforming/pull/259.
+
+				This just forces zone_id to exist for cloudflare_worker_route.
```
Comment (jacobbednarz) [inline internal/app/cf-terraforming/cmd/generate.go:647]: if this is the case, shouldn't it live around https://github.com/cloudflare/cf-terraforming/pull/348/files#diff-c8a271423d325a18f64e8eb426bae4c393315dfd1fbc9bad21916c9eead1647aR629?

File: internal/app/cf-terraforming/cmd/generate_test.go (hunk 1 above)
Comment (jafowler) [reply internal/app/cf-terraforming/cmd/generate_test.go:153]: Normally i'd agree, but due to the nature of the way Cloudflare-go does the worker api call, its necessary for both to exist so cf-terraforming can init Cloudflare-go with the accountID. 
https://github.com/cloudflare/cloudflare-go/blob/master/workers.go#L674 

Personally, i'd love to just default to `routes` in Cloudflare-go and it fixes all of this.

Comment (jacobbednarz) [reply internal/app/cf-terraforming/cmd/generate_test.go:153]: the flags are intended to be mutually exclusive so if this is working, it’s leveraging unintended and incorrect behaviour. there isn’t a case where you want to be using an account _and_ zone together; it’s one or the other. 

I’m happy to see this fixed in the Go library itself but we’ll just need to confirm the behaviour with the Workers team and the downstream libraries (like Terraform).

**Pull request closed without merging**